### PR TITLE
.github: Add ec2 information as a step

### PR DESCRIPTION
**Pull request by seemethere**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #63784

Also creates the common.yml.j2 file as a place to store common code
amongst the templates

Should look like: 
image

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D30490682](https://our.internmc.facebook.com/intern/diff/D30490682)

cc @ezyang @seemethere @malfet @walterddr @lg20987 @pytorch/pytorch-dev-infra

### CODE REVIEW

**Comment by seemethere** [comment]
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30490682).

File: .github/scripts/display_ec2_information.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+function get_ec2_metadata() {
+    category=$1
+    curl -fsSL "http://169.254.169.254/latest/meta-data/${category}"
```
Comment (driazati) [inline .github/scripts/display_ec2_information.sh:7]: Can you add a comment about where this is hosted?

**Review by driazati — APPROVED** [review]

**Review by malfet — APPROVED** [review]

File: .github/scripts/display_ec2_information.sh (hunk 1 above)
Comment (seemethere) [reply .github/scripts/display_ec2_information.sh:7]: Done

**Comment by seemethere** [comment]
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30490682).

**Pull request closed without merging**